File: cs/algorithms_in/recursive_selection_sort.cs
namespace algorithms_in;

public class RecursiveSelectionSort
{
    public static void recursiveSelectionSort(int[] array, int l, int r) {
        if (l < r) {
            int currentMin = array[l];
            int currentMinIndex = l;

            for (int i = l + 1; i < array.Length; i++) {
                if (currentMin > array[i]) {
                    currentMin = array[i];
                    currentMinIndex = i;
                }
            }
            array[currentMinIndex] = array[l];
            array[l] = currentMin;
            recursiveSelectionSort(array, l + 1, r);
        }
    }

}

File: rust/algorithms_in/src/recursive_selection_sort.rs
pub fn recursive_selection_sort(array: &mut [usize], l: usize, r: usize) {
    if l < r {
        let mut current_min: usize = array[l];
        let mut current_min_index: usize = l;
        for i in (l + 1)..array.len() {
            if current_min > array[i] {
                current_min = array[i];
                current_min_index = i
            }
        }
        array[current_min_index] = array[l];
        array[l] = current_min;
        recursive_selection_sort(array, l + 1, r);
    }
}